### PR TITLE
Feature: Change Theme Without Page Refresh

### PR DESCRIPTION
**Pull request by KevinMulhern**
Because
* Previously, we were refreshing the page when switching between themes. This was slow and inefficient.

This commit:
* Install Turbo Hotwire, but only enabled for the theme switcher and nothing else for now.
* Use Hotwire to turn on/off dark mode without page refresh
* Add theme value object to remove if conditions around theme display text and icons.
* Add a current_theme helper method to be used in views and controllers.

### CODE REVIEW

File: app/views/layouts/_dark_mode_styles.html.erb [hunk 1]
```diff
@@ -0,0 +1,3 @@
+<% if theme.dark_mode? %>
+  <%= stylesheet_link_tag 'dark-mode', media: 'all' %>
```
Comment (KevinMulhern) [inline app/views/layouts/_dark_mode_styles.html.erb:2]: It's worth mentioning we'll be getting rid of the dark mode stylesheet in a follow up PR in favour of using [Tailwinds dark variant](https://v2.tailwindcss.com/docs/dark-mode). Thats step two in [the plan](https://github.com/TheOdinProject/top-meta/issues/235) and this PR was getting big enough 😆

File: app/components/theme/switcher_component.rb [hunk 2]
```diff
@@ -1,25 +1,28 @@
 class Theme::SwitcherComponent < ViewComponent::Base
-  def initialize(theme:, mobile: false, icon_only: false)
-    @theme = theme
-    @mobile = mobile
-    @icon_only = icon_only
+  include Classy::Yaml::ComponentHelpers
```
Comment (ChargrilledChook) [inline app/components/theme/switcher_component.rb:2]: This is a neat idea. Always seemed a bit cumbersome to pass class names into a component

File: Gemfile [hunk 3]
```diff
@@ -79,3 +80,6 @@ end
 gem 'net-imap'
 gem 'net-pop'
 gem 'net-smtp'
+
+# Use Redis for Action Cable
```
Comment (ChargrilledChook) [inline Gemfile:84]: Surprised we didn't have this already, where were Sidekiq workers running before?

**Review by ChargrilledChook — APPROVED** [review]
Great idea mate, code looks really clean and it works a treat. Even when the dark mode style sheet is fetched for the first time it feels significant snappier  :muscle:

File: Gemfile [hunk 4]
```diff
@@ -42,6 +42,7 @@ gem 'sentry-ruby', '~> 5.4'
 gem 'sentry-sidekiq', '~> 5.4'
 gem 'shakapacker', '6.5.2'
 gem 'sidekiq', '~> 6.5'
+gem 'turbo-rails', '~> 1.3'
```
Comment (ChargrilledChook) [inline Gemfile:45]: Nice way to dip into Hotwire and test the waters :fire:

File: Gemfile (hunk 3 above)
Comment (KevinMulhern) [reply Gemfile:84]: Good question mate, Sidekiq doesn't need the Redis gem for most use cases https://github.com/mperham/sidekiq/wiki/Using-Redis. 

It connects directly to the default Redis port in development (localhost:6379). In production it looks for a `REDIS_URL` env var we have set up.

**Pull request merged**